lib/galaxy/workflow/modules.py

```diff
@@ -697,6 +700,63 @@ def get_runtime_inputs(self, **kwds):
         # TODO: Use a dict-based description from YAML tool source
         element = Element("param", name="input", label=self.label, type=parameter_type, optional=str(optional), **parameter_kwds)
         input = parameter_class(None, element)
+
+        # Retrieve possible runtime options for 'select' type inputs
+        if parameter_type == 'select' and connections:
+            static_options = []
+            for connection in connections:
+                tool_inputs = connection.input_step.module.tool.inputs
+                # Account for select options within conditionals
+                if connection.input_name not in tool_inputs and "|" in connection.input_name:
+                    param_name, case_name = connection.input_name.split("|", 1)
+                    if param_name in tool_inputs.data:
+                        param = tool_inputs.data[param_name]
+                    elif re.match('.+_\\d+', param_name):  # check for repeat label
```

Comment:
Using '\d+' to find repeat index as in 'rep_name_2'
```suggestion
                    elif re.match('.+_\d+', param_name):  # check for repeat label
```